core/drivers/scmi-msg/entry.c

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright (c) 2015-2019, Arm Limited and Contributors. All rights reserved.
+ * Copyright (c) 2019, Linaro Limited
+ */
+#include <speculation_barrier.h>
+#include <drivers/scmi-msg.h>
+#include <drivers/scmi.h>
+#include <trace.h>
+
+#include "base.h"
+#include "clock.h"
+#include "common.h"
+#include "reset_domain.h"
+
+scmi_msg_handler_t scmi_get_msg_handler(unsigned int message_id,
+					const scmi_msg_handler_t *handler_table,
+					size_t elt_count)
+{
+	/* Cast discards const qualifier to ease code readibility */
+	scmi_msg_handler_t *min = (scmi_msg_handler_t *)handler_table;
+	scmi_msg_handler_t *max = min + elt_count;
+	scmi_msg_handler_t *ptr = min + message_id;
+
+	return load_no_speculate_fail(ptr, min, max, NULL);
```

Comment:
I've added a more flexible mechanism to protect against speculation in https://github.com/OP-TEE/optee_os/pull/3484

Reply:
updating to recent [confine_array_index](https://github.com/OP-TEE/optee_os/pull/3649).